app/models/metrics/moves/counts_by_status_100_day.rb

```diff
@@ -0,0 +1,35 @@
+module Metrics
+  module Moves
+    class CountsByStatus100Day
+      include BaseMetric
+      include Moves
+
+      def initialize(supplier: nil)
+        setup_metric(
+          supplier: supplier,
+          label: 'Move counts by status for past 100 days',
+          file: 'counts_by_status_100_day',
+          interval: 6.hours, # we don't need to re-calculate this report very often (and it takes a while to calculate)
+          columns: {
+            name: 'status',
+            field: :itself,
+            values: Move.statuses.values << TOTAL,
+          },
+          rows: {
+            name: 'date',
+            field: :iso8601,
+            values: -> { ((Time.zone.today - 100.days)..(Time.zone.yesterday)).to_a.reverse },
```

Comment:
I've just been testing this locally, and this is running a query for every day, meaning we will make 100 requests per scope of supplier/moves, could we instead try and use one query for the date? 
An example:
```
moves_geoamey/counts_by_status_100_day/data.csv   (0.8ms)  SELECT COUNT(*) AS count_all, "moves"."status" AS moves_status FROM "moves" WHERE "moves"."supplier_id" = $1 AND "moves"."date" = $2 GROUP BY "moves"."status"  [["supplier_id", "bccf4d77-984d-560c-9ffd-9badbb9157ca"], ["date", "2021-01-26"]]
   (0.7ms)  SELECT COUNT(*) AS count_all, "moves"."status" AS moves_status FROM "moves" WHERE "moves"."supplier_id" = $1 AND "moves"."date" = $2 GROUP BY "moves"."status"  [["supplier_id", "bccf4d77-984d-560c-9ffd-9badbb9157ca"], ["date", "2021-01-25"]]
   (0.6ms)  SELECT COUNT(*) AS count_all, "moves"."status" AS moves_status FROM "moves" WHERE "moves"."supplier_id" = $1 AND "moves"."date" = $2 GROUP BY "moves"."status"  [["supplier_id", "bccf4d77-984d-560c-9ffd-9badbb9157ca"], ["date", "2021-01-24"]]
   (0.7ms)  SELECT COUNT(*) AS count_all, "moves"."status" AS moves_status FROM "moves" WHERE "moves"."supplier_id" = $1 AND "moves"."date" = $2 GROUP BY "moves"."status"  [["supplier_id", "bccf4d77-984d-560c-9ffd-9badbb9157ca"], ["date", "2021-01-23"]]
   (0.9ms)  SELECT COUNT(*) AS count_all, "moves"."status" AS moves_status FROM "moves" WHERE "moves"."supplier_id" = $1 AND "moves"."date" = $2 GROUP BY "moves"."status"  [["supplier_id", "bccf4d77-984d-560c-9ffd-9badbb9157ca"], ["date", "2021-01-22"]]
   (0.6ms)  SELECT COUNT(*) AS count_all, "moves"."status" AS moves_status FROM "moves" WHERE "moves"."supplier_id" = $1 AND "moves"."date" = $2 GROUP BY "moves"."status"  [["supplier_id", "bccf4d77-984d-560c-9ffd-9badbb9157ca"], ["date", "2021-01-21"]]
   ....
```

Reply:
A count grouped by date (maybe truncating a date/time if needed) should do that, but might mean dropping to lower level SQL, or perhaps using Arel with `pluck` - that will return essentially a huge array that you can pick through as needed. I use a technique like this for the location free spaces calculation and it's really performant: https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/models/population.rb#L87

Reply:
now improved to use group_by(:status, :date)

Reply:
Nice 👍